batcher/treasury.mligo

```diff
@@ -136,178 +123,229 @@ module Utils = struct
+      | (MINUS,PLUS) -> false
+      | (EXACT,PLUS) -> false
+      | (_,_) -> true
+
+  let was_in_clearing_for_sell
+   (clearing_tolerance: tolerance)
+   (order_tolerance: tolerance) : bool =
+      match (order_tolerance, clearing_tolerance) with
+      | (EXACT,MINUS) -> false
+      | (PLUS,MINUS) -> false
+      | (MINUS,EXACT) -> true
+      | (PLUS,EXACT) -> false
+      | (MINUS,PLUS) -> true
+      | (EXACT,PLUS) -> true
+      | (_,_) -> true
+
+
+  let was_in_clearing
+    (order:order)
+    (clearing: clearing) : bool =
+    let order_tolerance = order.tolerance in
+    let order_side = order.side in
+    let clearing_tolerance = clearing.clearing_tolerance in
+    match order_side with
+    | BUY -> was_in_clearing_for_buy clearing_tolerance order_tolerance
+    | SELL -> was_in_clearing_for_sell clearing_tolerance order_tolerance
+
+  let get_clearing
+    (batch: batch) : clearing option =
+    match batch.status with
+    | Cleared { at = _ ; clearing = c; rate = _ } -> Some c
+    | _ -> None
+
+
+  let get_clearing_volume
+    (clearing:clearing) : nat =
+    match clearing.clearing_tolerance with
+    | MINUS -> clearing.clearing_volumes.minus
+    | EXACT -> clearing.clearing_volumes.exact
+    | PLUS -> clearing.clearing_volumes.plus
+
+  let get_cleared_sell_side_payout
+    (swap:swap)
+    (clearing:clearing) : token_amount list =
+    let f_sell_side_actual_volume = Float.new (int clearing.prorata_equivalence.sell_side_actual_volume) 0 in
+    let f_amount = Float.new (int swap.from.amount) 0 in
+    let prorata_allocation = Float.div f_amount f_sell_side_actual_volume in
+    let f_buy_side_clearing_volume = Float.new (int (get_clearing_volume clearing)) 0 in
+    let payout = Float.mul prorata_allocation f_buy_side_clearing_volume in
+    let payout_equiv = Float.mul clearing.clearing_rate.rate payout in
+    let remaining = Float.sub f_amount payout_equiv in
+    let fill_payout: token_amount = {
+      token = swap.to;
+      amount = Math.get_rounded_number payout;
+    } in
+    if Float.gt remaining (Float.new 0 0) then
+      let token_rem : token_amount = {
+         token = swap.from.token;
+         amount = Math.get_rounded_number remaining;
+      } in
+      [ fill_payout; token_rem ]
+    else
+      [ fill_payout ]
+
+  let get_cleared_buy_side_payout
+    (swap:swap)
+    (clearing:clearing) : token_amount list =
+    let f_buy_side_actual_volume = Float.new (int clearing.prorata_equivalence.buy_side_actual_volume) 0 in
+    let f_amount = Float.new (int swap.from.amount) 0 in
+    let prorata_allocation = Float.div f_amount f_buy_side_actual_volume in
+    let f_buy_side_clearing_volume = Float.new (int (get_clearing_volume clearing)) 0 in
+    let f_sell_side_clearing_volume = Float.mul clearing.clearing_rate.rate f_buy_side_clearing_volume in
+    let payout = Float.mul prorata_allocation f_sell_side_clearing_volume in
+    let payout_equiv = Float.div clearing.clearing_rate.rate payout in
```

Comment:
I think this line should be `let payout_equiv = Float.div payout clearing.clearing_rate.rate`.

Reply:
Fixed